### DISCUSSION TITLE
TheoryData get wrong

### DISCUSSION THREAD

**Discussion by Aedjis** [xunit/xunit#2409] [General]
I have experimented a issue using TheoryData.

Here is the TheoryData

```
public static TheoryData<string, DateTimeOffset?, HttpStatusCode> Cases
{
	get
	{
		return new TheoryData<string, DateTimeOffset?, HttpStatusCode>()
		{
			{ "Rivoli", null, HttpStatusCode.NotFound },
			{ string.Empty, null, HttpStatusCode.BadRequest },
			{ "invalid", null, HttpStatusCode.BadRequest },
			{ "Louise", DateTimeOffset.UtcNow.Subtract(TimeSpan.FromDays(1)), HttpStatusCode.BadRequest },
			{ "Louise,", null, HttpStatusCode.OK },
			{ "Louise", DateTimeOffset.UtcNow, HttpStatusCode.OK }
		};
	}
}
```

the issue is some time the DateTimeOffset.UtcNow will not be the current UtcNow but the utcNow of 15 minute ago. 
I think it is because he as been serialized (the desplay in the Test explorer is one per line, this is the reason why I thinks he as been serialized).

**Comment by bradwilson** [comment]
Your understanding is correct. We talk about "theory data stability" in the context of Visual Studio here: https://xunit.net/faq/theory-data-stability-in-vs

You don't say you're using Visual Studio, but AFAIK it's the only one that uses the serialized test case system, so I'm guessing you are.

If you need data that is always fresh and never serialize, you should disable theory data pre-enumeration in that particular test. What you'll lose is the ability to run an individual data row in the VS UI (you'll have to run the whole test case). Just add `DisableDiscoveryEnumeration = true` to your `[MemberData]` attribute.
